.github/workflows/CI.yml

```diff
@@ -3,39 +3,63 @@ name: CI
 on:
   push:
     branches:
-      - main
-      - master
+      - "main"
+    paths:
+      - '.github/workflows/CI.yml'
+      - 'ext/**'
+      - 'src/**'
+      - 'test/**'
+      - 'Project.toml'
   pull_request:
+    paths:
+      - '.github/workflows/CI.yml'
+      - 'ext/**'
+      - 'src/**'
+      - 'test/**'
+      - 'Project.toml'
+  release:
   workflow_dispatch:
 
+concurrency:
+  # Skip intermediate builds: always.
+  # Cancel intermediate builds: always.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+# needed to allow julia-actions/cache to delete old caches that it has created
+permissions:
+  actions: write
+  contents: read
+
 jobs:
   test:
     name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
     runs-on: ${{ matrix.os }}
+    timeout-minutes: 30
     strategy:
       fail-fast: false
       matrix:
         version:
-          - '1.10'  # Latest release
+          - "min"
```

Comment:
For what is worth, tests pass on Julia v1.11 as well.  I know Oceananigans isn't fully compatible with v1.11 yet, but I believe testing multiple versions for a downstream package would still be a good idea, not to get stuck forever to an old version, and this may help keeping Oceananigans up to date without having to test the full upstream package with that version.

Reply:
So I can add `"1"`?

Reply:
lets do it

Reply:
we'll find out if it doesn't work...